core/src/services/azfile/core.rs

```diff
@@ -420,6 +420,21 @@ impl AzfileCore {
             p = get_parent(p);
             dirs.push_front(p);
         }
+
+        let mut pop_dir_count = dirs.len() as u8;
+        for dir in dirs.iter().rev() {
+            let resp = self.azfile_get_path_properties(dir).await?;
+            if resp.status() == StatusCode::NOT_FOUND {
+                pop_dir_count -= 1;
+                continue;
+            }
+            break;
+        }
+
+        for _ in 0..pop_dir_count {
+            dirs.pop_front();
+        }
+
```

Comment:
We can use dirs.iter().skip() to avoid calling `pop_front()` in a for loop.

Reply:
`let dirs = dirs.iter().skip(pop_dir_count as usize).collect::<Vec<_>>();`

1. use `Vec` or `Vecdeque` ?
2. pop_dir_count type: `u8` or `usize` which is better ?